src/marketplacePlugin.ts

```diff
@@ -140,6 +145,102 @@ export class Web3MarketplacePlugin extends Web3PluginBase {
 
 		return receipt;
 	}
+
+	/**
+	 * Gets all items in a marketplace.
+	 * Path: https://api.gaming.chainsafe.io/v1/projects/{projectID}/marketplaces/{marketplaceID}/items
+	 *
+	 * @param {string} marketplaceId - Marketplace ID to query.
+	 * @param {string} customProjectId - Project ID to query (optional).
+	 * @returns {Promise<marketplaceItemsResult>} - Returns the marketplace items response object.
+	 */
+	public async getMarketplaceItems(
+		marketplaceId: string,
+		customProjectId: string,
+	): Promise<marketplaceItemsResult> {
+		// Define the API base URL
+		const baseUrl = 'https://api.gaming.chainsafe.io/v1';
```

Comment:
I'll suggest to make baseurl as class member and set it via constructor. also have 'https://api.gaming.chainsafe.io/v1' default value of that via constructor, as its used in multiple places in this class.